src/main/java/org/mastodon/mamut/tomancak/resolve/LocateTagsFrame.java

```diff
@@ -0,0 +1,503 @@
+		updateSpotToRows( this.rows );
+		tableModel.fireChange();
+	}
+
+	private static List< Row > getRows( final Model model, final TagSetStructure.TagSet tagSet )
+	{
+		final ModelGraph graph = model.getGraph();
+		final RefSet< Spot > roots = RootFinder.getRoots( graph );
+		final DepthFirstIterator< Spot, Link > depthFirstIterator = new DepthFirstIterator<>( graph );
+		final ObjTagMap< Spot, TagSetStructure.Tag > spotToTag = model.getTagSetModel().getVertexTags().tags( tagSet );
+		final Spot ref = graph.vertexRef();
+		final List< Row > rows = new ArrayList<>();
+		for ( final Spot root : roots )
+		{
+			depthFirstIterator.reset( root );
+			while ( depthFirstIterator.hasNext() )
+			{
+				final Spot spot = depthFirstIterator.next();
+				final TagSetStructure.Tag tag = spotToTag.get( spot );
+				if ( !entryPoint( spotToTag, spot, tag, ref ) )
+					continue;
+				rows.add( new Row( graph, tag, spot, root.getLabel() ) );
+			}
+		}
+		// TODO fix sorting
+		rows.sort( Comparator.comparing( Row::toString ) );
+		return rows;
+	}
+
+	private void updateSpotToRows( final List< Row > rows )
+	{
+		spotToRow.clear();
+		for ( final Row row : rows )
+			spotToRow.put( row.spot, row );
+	}
+
+
+	private class MyGraphListener implements GraphListener< Spot, Link >
+	{
+		@Override
+		public void graphRebuilt()
+		{
+			SwingUtilities.invokeLater( LocateTagsFrame.this::fillList );
+		}
+
+		@Override
+		public void vertexAdded( final Spot spot )
+		{
+
+		}
+
+		@Override
+		public void vertexRemoved( final Spot spot )
+		{
+			final Row row = spotToRow.get( spot );
+			if ( row != null )
+			{
+				rows.remove( row );
+				spotToRow.remove( spot );
+				tableModel.fireChange();
+			}
+		}
+
+		@Override
+		public void edgeAdded( final Link link )
+		{
+
+		}
+
+		@Override
+		public void edgeRemoved( final Link link )
+		{
+
+		}
+	}
+
+
+	private class MyTableModel implements TableModel
+	{
+
+		private final List< TableModelListener > listeners = new CopyOnWriteArrayList<>();
+
+		private final List< String > columns = Arrays.asList( "time point", "tag", "", "track", "spot" );
+
+		private final List< Class< ? > > classes = Arrays.asList( Integer.class, String.class, Color.class, String.class, String.class );
+
+		@Override
+		public int getRowCount()
+		{
+			return rows.size();
+		}
+
+		@Override
+		public int getColumnCount()
+		{
+			return columns.size();
+		}
+
+		@Override
+		public String getColumnName( final int columnIndex )
+		{
+			return columns.get( columnIndex );
+		}
+
+		@Override
+		public Class< ? > getColumnClass( final int columnIndex )
+		{
+			return classes.get( columnIndex );
+		}
+
+		@Override
+		public boolean isCellEditable( final int rowIndex, final int columnIndex )
+		{
+			return false;
+		}
+
+		@Override
+		public Object getValueAt( final int rowIndex, final int columnIndex )
+		{
+			final Row row = rows.get( rowIndex );
+			switch ( columnIndex )
```

Comment:
Why would you need one? What should be the default statement?

Reply:
It is not really needed in this case, since the `return null` is there immediately after the `switch` statement.

The `return null` could be the `default` case of the `switch` statement, but you can leave it as it is. The result is the same.